lib/data.dart

```diff
@@ -22,27 +22,35 @@ class Student {
   String name;
   String yearId;
 
-  // TODO: add issues
-  // TODO: add feedback, up/down votes, representatives, comments and stuff that we will  have time for
   Student(this.id, this.name, this.yearId);
 }
 
 class TA {
   String id;
   String name;
 
-  // TODO: add issues
-  // TODO: add rating
   TA(this.name, this.id);
 }
 
 class TaCourse {
   String taId;
   String courseId;
-  double rating;
   String docId;
+  double avgRating;
+  List<int> ratings;
+
+  calculateAvgRating() {
+    int ratingsCount = ratings.reduce((value, element) => value + element);
+    int sumRatings = 0;
+    for (int i = 0; i < 5; i += 1) {
+      sumRatings += (i + 1) * ratings[i];
+    }
+    avgRating = (sumRatings / (ratingsCount * 5)) * 5;
```

Comment:
`avgRating = ratingCount/ratings.length` or I don't understand what do you want🤔 And I believe some return statement is required

Reply:
No return is required. It is saved as a class member.
I will double-check the math.

Reply:
Fixed.